examples/tree_1d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,100 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the shallow water equations
+
+equations = ShallowWaterEquations1D(gravity_constant=9.812)
+cfl = .3
+
+function initial_condition_well_balanced_CN(x, t, equations:: ShallowWaterEquations1D)
+  v = 0
+  b = sin(4*pi*x[1]) + 3
+
+  if x[1] >= 0.5
+    b = sin(4*pi*x[1]) + 1
+  end
+
+  H = max(b, 2.5)
+
+  if x[1] >= 0.5
+    H = max(b, 1.5)
+  end
+
+  H = max(H, b + equations.threshold_limiter)
```

Comment:
Add small comment about what the `equations.threshold_limiter` is and why it is needed.